lib/streamlit/MediaFileManager.py

```diff
@@ -26,6 +28,8 @@
 
 STATIC_MEDIA_ENDPOINT = "/media"
 
+# Seconds to keep media files that have been obsoleted by replacement-in-place.
+KEEP_DELAY = 2  
```

Comment:
Why is this needed? I don't remember...

In general when I see this kind of code I worry about possible race conditions. So I'd like to remember the rationale for this, and if it makes sense we should add a comment explaining it here for posterity.

Reply:
Not something we talked about, which is why i wanted to make sure you saw this PR.  

In rapid-fire replacement scenarios, e.g. creating a video from repeated calls to the same `st.image` element using a new different image, the image gets replaced before it is requested by the browser.

So i introduced a 2-second (somewhat arbitrary) delay controlled by a timestamp on the media file.  

I was trying to avoid a solution like this since you have to iterate over the whole list of files to weed them, but I'm not sure of a better way to fix this issue at this time.